### PR TITLE
Announcement computer tgui

### PR DESCRIPTION
**Pull request by Valtsu0**
[UI]
## About the PR 
image
image
## Why's this needed? 
tgui good

## Changelog 

```changelog
(u)Valtsu0
(+)Announcement computers now use tgui
```

### CODE REVIEW

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
The transmit button and message only update when you click away, meaning you can click a visually disabled button. Best to set the status of the button and the "input message" text on the frontend based on whether the textbox contains a message. See how `TextInputModal` does this for details (`useSharedState`)

**Comment by Valtsu0** [comment]
Actually, i see of no reason why the message box should be shared between users. Made it local

**Review by Tarmunora — APPROVED** [review]
light theme checks out

**Comment by Cherman0** [comment]
There is a maximum character limit to an announcement that is currently not visible in the UI, leading people to sometimes transmit cut-off messages. Would probably be good to add a "message length / max length" display so people can know about that.

**Comment by Valtsu0** [comment]
> There is a maximum character limit to an announcement that is currently not visible in the UI, leading people to sometimes transmit cut-off messages. Would probably be good to add a "message length / max length" display so people can know about that.

image
well it was kinda in the pr, will fix that

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
The previous interface let you input a message and then hit transmit later, this one clears the message if you close the interface and sends if you hit enter. I think being able to input a message to transmit later is useful QoL and hitting enter to send is mostly just going to result in accidental transmissions as people try to insert newlines.
Other than that, this looks great!

File: tgui/packages/tgui/interfaces/AnnouncementComputer/automatic.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { Input, Section, Stack } from '../../components';
+import { AnnouncementCompData } from './data';
+
+export interface AutomaticAnnouncementData {
+  arrivalalert: string;
+}
+
+export const AutomaticAnnouncement = (_props, context) => {
+  const { act, data } = useBackend<AnnouncementCompData>(context);
+  // Extract `health` and `color` variables from the `data` object.
```
Comment (TobleroneSwordfish) [inline tgui/packages/tgui/interfaces/AnnouncementComputer/automatic.tsx:18]:
```suggestion
```

File: tgui/packages/tgui/interfaces/AnnouncementComputer/manual.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+type Status = {
+  canTransmit: boolean;
+  text: string;
+  color: string;
+};
+
+export const ManualAnnouncement = (_props, context) => {
+  const { act, data } = useBackend<AnnouncementCompData>(context);
+  // Extract `health` and `color` variables from the `data` object.
```
Comment (TobleroneSwordfish) [inline tgui/packages/tgui/interfaces/AnnouncementComputer/manual.tsx:21]:
```suggestion
```

File: tgui/packages/tgui/interfaces/AnnouncementComputer/data.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/**
```
Comment (TobleroneSwordfish) [inline tgui/packages/tgui/interfaces/AnnouncementComputer/data.tsx:1]: This file should be called `type.ts` to be consistent with other interfaces.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
The UI won't register a card if it's inserted when the UI is closed, and re-opening the UI sets the caret position to the start of the text.

**Comment by Valtsu0** [comment]
> re-opening the UI sets the caret position to the start of the text.

That seems to be an issue with tgui/inferno. Changing your character's name has the same issue.

clearing the message after transmitting should help a bit

**Review by TobleroneSwordfish — APPROVED** [review]

**Pull request merged**